pt.py

```diff
@@ -157,9 +158,14 @@ def __init__(self):
 
     def lazy_init(self):
         # Get quick access to physical memory.
-        proc = os.popen("pgrep qemu-system")
-        self.pid = int(proc.read().strip(), 10)
-        proc.close()
+
+        # Technically `info cpus` gives us a thread id, but this is sufficient for use with /proc/[tid]/mem
+        cpus = gdb.execute("monitor info cpus", to_string = True)
+        m = re.search('thread_id=(\d+)', cpus)
+        if not m:
+            raise Exception(f"Could not find qemu pid")
```

Comment:
Could you please add a fallback option to use pgrep?

The problems I see with this approach:
- qemu and gdb may be in different pid namespaces
- the "monitor info cpus" may return different output or may not be available across different qemu versions.

Something like this:
1) Try to  get info via "monitor info cpus". If success, check that process actually exists and it's qemu
2) If fails, print message and print  info via pgrep qemu-system.

Would this make sense?